lib/microformats/parser_core.rb

```diff
@@ -347,5 +347,34 @@ def value_title_classes(element)
         html_class =~ VALUE_TITLE_CLASS_REG_EXP
       end
     end
+
+    def render_and_strip(data)
+      new_doc = Nokogiri::HTML(data)
+      new_doc.xpath('//script').remove
+      new_doc.xpath('//style').remove
+      new_doc.text.strip
+    end
+
+    def render_text(node, base: nil)
+      render_text_and_replace_images(node, base: base)
+    end
+
+    def render_text_and_replace_images(in_node, base: nil)
+      new_doc = Nokogiri::HTML(in_node.inner_html)
+      new_doc.xpath('//script').remove
+      new_doc.xpath('//style').remove
+
+      new_doc.traverse do |node|
+        if node.name == 'img' && !node.attribute('alt').nil?
+          node.replace(node.attribute('alt').value.to_s)
+        elsif node.name == 'img' && !node.attribute('src').nil?
+          absolute_url = Microformats::AbsoluteUri.new(node.attribute('src').value.to_s, base: base).absolutize
+
+          node.replace(' ' + absolute_url + ' ')
+        end
+      end
+
+      new_doc.text.strip
+    end
```

Comment:
Forgot to note in my initial comment: This refactoring uses `doc.css('img')` to narrow down the result set rather than traversing the entire `doc` for `img` elements.

Reply:
definitely cleans things up a bit,b tu i very much dislike using `&& next unless` just to get around rubocop complaints of function length.  conciseness does not trump readability.  Same reason i almost never use single line conditional modifiers normally.  I'm less opposed to those however